lib/root_route_constraint.rb

```diff
@@ -0,0 +1,13 @@
+class RootRouteConstraint
+  def initialize
+    @current = Conference.where('end_date >= ?', Date.current).reorder(start_date: :asc)
+  end
+
+  def matches?(*)
+    if @current.present? && @current.first.splashpage.present?
+      @current.count == 1 && @current.first.splashpage.public?
```

Comment:
yes if `@current.count == 1 && @current.first.splashpage.public?` is true then it return true else false, but first there need to check if `@current` and `@current.first.splashpage` are not nil.